extensions/amp-base-carousel/0.1/amp-base-carousel.js

```diff
@@ -449,7 +449,7 @@ class AmpCarousel extends AMP.BaseElement {
       'goToSlide',
       (actionInvocation) => {
         const {args, trust} = actionInvocation;
-        this.carousel_.goToSlide(args['index'] || -1, {
+        this.carousel_.goToSlide(Number(args['index'] || 0), {
```

Comment:
```suggestion
        this.carousel_.goToSlide(Number(args['index']) || -1, {
```

We want to cast `args['index']` as a `Number`, not `args['index'] || 0`, since this could have the side effect of `goToSlide('foo')` also resulting in `NaN`. 

I would also recommend keeping the default index as `-1` instead of `0` so that invalid inputs do not default to resetting the slide index to `0`.